### DISCUSSION TITLE
"Reset all bindings in section" is part of key configuration filter text

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#14877] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/16479013/135079915-941dfb6b-0083-4b69-9f39-db8bf71960a6.png)

This text shouldn't be searchable. Makes it hard to navigate sections (as there are many "empty" sections) if your search query contains word similar to "Reset all bindings in section".

**Comment by Susko3** [comment, marked as answer]
The button should probably be shown if and only if the whole section is visible. Otherwise, the button might be misleading, as even filtered-out keybinds are reset.

**Reply by Susko3** [reply]
This can be achieved with a simple change in `ResetButton`:

```diff
diff --git a/osu.Game/Overlays/Settings/Sections/Input/KeyBindingsSubsection.cs b/osu.Game/Overlays/Settings/Sections/Input/KeyBindingsSubsection.cs
index fae0318359..f25f3816aa 100644
--- a/osu.Game/Overlays/Settings/Sections/Input/KeyBindingsSubsection.cs
+++ b/osu.Game/Overlays/Settings/Sections/Input/KeyBindingsSubsection.cs
@@ -75,5 +75,7 @@ private void load()

             Content.CornerRadius = 5;
         }
+
+        public override IEnumerable<string> FilterTerms => Enumerable.Empty<string>();
     }
 }
```

**Reply by bdach** [reply]
I wouldn't be opposed to that solution, personally.

**Reply by peppy** [reply]
Just for the record, this was resolved via https://github.com/ppy/osu/pull/14937.
